lib/web_ui/dev/build.dart

```diff
@@ -68,78 +75,66 @@ class BuildCommand extends Command<bool> with ArgUtils<bool> {
 /// Not safe to interrupt as it may leave the `out/` directory in a corrupted
 /// state. GN is pretty quick though, so it's OK to not support interruption.
 class GnPipelineStep extends ProcessStep {
-  GnPipelineStep(this.buildCanvasKit);
+  GnPipelineStep({this.target = 'engine'})
+      : assert(target == 'engine' || target == 'sdk');
 
   @override
   String get description => 'gn';
 
   @override
   bool get isSafeToInterrupt => false;
 
-  /// Whether or not to build CanvasKit.
-  final bool buildCanvasKit;
+  /// The target to build with gn.
+  ///
+  /// Acceptable values: engine, canvaskit
+  final String target;
 
   @override
   Future<ProcessManager> createProcess() {
     print('Running gn...');
-    Future<ProcessManager> gnProcess = startProcess(
-      path.join(environment.flutterDirectory.path, 'tools', 'gn'),
-      <String>[
+    final List<String> gnArgs = <String>[];
+    if (target == 'engine') {
+      gnArgs.addAll(<String>[
         '--unopt',
         if (Platform.isMacOS) '--xcode-symlinks',
         '--full-dart-sdk',
-      ],
-    );
-    if (buildCanvasKit) {
-      gnProcess = gnProcess.then((_) {
-        return startProcess(
-          path.join(environment.flutterDirectory.path, 'tools', 'gn'),
-          <String>[
-            '--wasm',
-          ],
-        );
-      });
+      ]);
+    } else if (target == 'canvaskit') {
+      gnArgs.add('--wasm');
+    } else {
+      throw StateError('Target was not engine or canvaskit: $target');
     }
-    return gnProcess;
+    return startProcess(
+      path.join(environment.flutterDirectory.path, 'tools', 'gn'),
+      gnArgs,
+    );
   }
 }
 
 /// Runs `autoninja`.
 ///
 /// Can be safely interrupted.
 class NinjaPipelineStep extends ProcessStep {
-  NinjaPipelineStep(this.buildCanvasKit);
+  NinjaPipelineStep({required this.target});
 
   @override
   String get description => 'ninja';
 
   @override
   bool get isSafeToInterrupt => true;
 
-  /// Whether or not to build CanvasKit.
-  final bool buildCanvasKit;
+  /// The target directory to build.
+  final Directory target;
 
   @override
   Future<ProcessManager> createProcess() {
     print('Running autoninja...');
-    Future<ProcessManager> ninjaProcess = startProcess(
+    return startProcess(
       'autoninja',
       <String>[
         '-C',
         environment.hostDebugUnoptDir.path,
```

Comment:
Do we need to pass `target` here? I'm actually not seeing where the `target` field is used :thinking:

Reply:
Done